aptos-move/framework/src/natives/cryptography/bls12381.rs

```diff
@@ -622,23 +624,31 @@ pub fn native_generate_keys(
 
 #[cfg(feature = "testing")]
 pub fn native_sign(
-    _context: &mut SafeNativeContext,
+    context: &mut SafeNativeContext,
     _ty_args: Vec<Type>,
     mut arguments: VecDeque<Value>,
 ) -> SafeNativeResult<SmallVec<[Value; 1]>> {
     let msg = safely_pop_arg!(arguments, Vec<u8>);
     let sk_bytes = safely_pop_arg!(arguments, Vec<u8>);
+
+    context.charge(
+        BLS12381_TEST_ONLY_SIGN_BASE
+            + BLS12381_TEST_ONLY_SIGN_PER_BYTE_MSG * NumBytes::new(msg.len() as u64),
+    )?;
```

Comment:
@alinush does this formula look correct to you? 

Later we'll need to do the same thing for other test only natives.

Reply:
Yes, that looks good. The message is first hashed (linear gas formula) and then some costant time operations are performed on that hash.

Reply:
For `t`-out-of-`n` MultiEd25519 though... there are `t` signatures being computed though. So you will need to account for that.